app/code/Magento/Catalog/Controller/Category/View.php

```diff
@@ -189,6 +188,15 @@ protected function _initCategory()
         $this->_catalogSession->setLastVisitedCategoryId($category->getId());
         $this->_coreRegistry->register('current_category', $category);
         $this->toolbarMemorizer->memorizeParams();
+        // If category is used as homepage - set its url as alias for layered navigation links
+        if ($this->isHomepage() &&
+            !$this->_request->getAlias(UrlInterface::REWRITE_REQUEST_PATH_ALIAS)) {
+            $this->_request->setAlias(
+                UrlInterface::REWRITE_REQUEST_PATH_ALIAS,
+                str_replace($this->_url->getBaseUrl(), '', $category->getUrl())
+            );
+        }
```

Comment:
It would be good to extract this block to a private method

Reply:
@sivaschenko done